app/(public)/getProducts/page.tsx

```diff
@@ -0,0 +1,117 @@
+                    quantity: quantity
+                })
+            })
+            if (!response.ok) {
+                throw new Error(`HTTP error! status: ${response.status}`);
+            }
+            const data = await response.json();
+            console.log(data)
+            setProducts(data.rows);
+        } catch (error) {
+            console.error('An error occurred while fetching the data.', error);
+        }
+    }
+
+    async function handleGet(e: React.MouseEvent<HTMLButtonElement>) {
+        e.preventDefault();
+        try {
+            const response = await fetch(`/api/getProducts?itemCode=${realItemCode}`, {
+                method: 'GET',
+                headers: {
+                    'Content-Type': 'application/json'
+                }
+            })
+            if (!response.ok) {
+                throw new Error(`HTTP error! status: ${response.status}`);
+            }
+            const data = await response.json();
+            console.log(data)
+            setProducts(data);
+        } catch (error) {
+            console.error('An error occurred while fetching the data.', error);
+        }
+    }
+
+    async function handleGetAll(e: React.MouseEvent<HTMLButtonElement>) {
+        e.preventDefault();
+        try {
+            const response = await fetch(`/api/getProducts`, {
+                method: 'GET',
+                headers: {
+                    'Content-Type': 'application/json'
+                }
+            })
+            if (!response.ok) {
+                throw new Error(`HTTP error! status: ${response.status}`);
+            }
+            const data = await response.json();
+            console.log(data)
+            setProducts(data);
+        } catch (error) {
+            console.error('An error occurred while fetching the data.', error);
+        }
+    }
+
+
+    return (
+        <div className="flex flex-wrap justify-center place-items-center text-white">
+            <div className="w-full flex flex-col items-center justify-center mt-20">
+                <form className="h-40 w-[700px] border border-black flex justify-center items-center text-black my-2" onSubmit={handlePost}>
+                    <input className="w-40 h-10 border pl-2 border-black mx-2" type="text" onChange={(e) => setProductName(e.target.value)} placeholder="Product Name" />
+                    <input className="w-40 h-10 border pl-2 border-black mx-2" type="text" onChange={(e) => setItemCode(e.target.value)} placeholder="Item Code" />
+                    <input className="w-40 h-10 border pl-2 border-black mx-2" type="text" onChange={(e) => setQuantity(parseInt(e.target.value))} placeholder="Quantity" />
+                    <button className={`w-20 h-10 bg-black text-white rounded-sm ${itemCode !== '' && productName !== '' && quantity > 0 ? 'cursor-pointer' : 'cursor-not-allowed'}`} disabled={itemCode === '' && productName === '' && quantity <= 0} type="submit">POST</button>
```

Comment:
A validation check for fields? If needed, separate it in a variable for readability